plugin/tests/run-pass/while_let.rs

```diff
@@ -0,0 +1,25 @@
+#![feature(plugin)]
+#![plugin(mutagen_plugin)]
+#![feature(custom_attribute)]
+
+extern crate mutagen;
+
+#[mutate]
+fn main() {
+    test_optional();
+}
+
+
+fn test_optional() {
```

Comment:
You need to add the `#[mutate]` annotation on this function. Otherwise, you are only mutating the `main` function.